bundlepath.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmrepo
+
+import (
+	"os"
+	"path/filepath"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+)
+
+// BundlePath is used to access a bundle at a given path.
+type BundlePath interface {
+	// Bundle returns the bundle represented by this path.
+	Bundle() (charm.Bundle, *charm.URL)
+}
+
+// bundlePath represents a path (eg local directory) containing a
+// single bundle.
+type bundlePath struct {
+	Name       string
+	BundleInfo charm.Bundle
+}
+
+var _ BundlePath = (*bundlePath)(nil)
+
+// NewBundlePath creates and returns an instance used to
+// open a bundle located at the given path.
+func NewBundlePath(path string) (BundlePath, error) {
```

Comment:
I don't see how this approach helps. If you need to extend it, you'll need to change the BundlePath.Bundle method, which will still be a breaking change. If you really wanted the return from NewBundlePath to be extensible it should return a struct to which you could add members, not an interface, but I think it's reasonable to just return what you need to return (the bundle and the URL) directly.

If we need something that returns more, we can always just add another function later and maintain backward compatibility that way.